src/lib/stores/schedules.ts

```diff
@@ -1,148 +1,162 @@
-    label: 'Minute',
-    required: false,
-  },
-  second: {
-    key: 'schedule.spec.calendar.second',
-    label: 'Second',
-    required: false,
-  },
-
-  // TODO: Post Alpha, add support of additional fields.
-  // "startTime": "2022-07-04T03:18:59.668Z",
-  // "endTime": "2022-07-04T03:18:59.668Z",
-  // "jitter": "string",
-  // "timezoneName": "string",
-  // "timezoneData": "string"
+const setBodySpec = (
+  body: DescribeFullSchedule,
+  spec: ScheduleSpecParameters,
+  presets: SchedulePresetsParameters,
+) => {
+  const { hour, minute, second, phase, cronString } = spec;
+  const { preset, months, days, daysOfMonth, daysOfWeek } = presets;
+  if (preset === 'string') {
+    // Add the cronString as a comment to the cronString to view it for frequency
+    const cronStringWithComment = `${cronString}#${cronString}`;
+    body.schedule.spec.cronString = [cronStringWithComment];
+    body.schedule.spec.calendar = [];
+    body.schedule.spec.interval = [];
+  } else if (preset === 'interval') {
+    const interval = timeToInterval(days, hour, minute, second);
+    body.schedule.spec.interval = [{ interval, phase: phase || '0s' }];
+    body.schedule.spec.cronString = [];
+    body.schedule.spec.calendar = [];
+  } else {
+    const { month, dayOfMonth, dayOfWeek } = convertDaysAndMonths({
+      months,
+      daysOfMonth,
+      daysOfWeek,
+    });
+    const comment = calendarToComment({
+      preset,
+      month,
+      dayOfMonth,
+      dayOfWeek,
+      hour,
+      minute,
+      second,
+    });
+    body.schedule.spec.calendar = [
+      {
+        year: '*',
+        month: preset === 'month' ? month : '',
+        dayOfMonth: preset === 'month' ? dayOfMonth : '',
+        dayOfWeek: preset === 'week' ? dayOfWeek : '',
+        hour,
+        minute,
+        second,
+        comment,
+      },
+    ];
+    body.schedule.spec.interval = [];
+    body.schedule.spec.cronString = [];
+  }
 };
 
-export const submitScheduleForm = async (
-  form: Form,
-  namespace: string,
-): Promise<void> => {
-  const values = form.values;
-  const body = {
-    schedule_id: '',
+export const submitCreateSchedule = async ({
+  action,
+  spec,
+  presets,
+}: ScheduleParameters): Promise<void> => {
+  const { namespace, name, workflowId, workflowType, taskQueue } = action;
+  const body: DescribeFullSchedule = {
+    schedule_id: name,
     schedule: {
       spec: {
         calendar: [],
         interval: [],
+        cronString: [],
       },
       action: {
         startWorkflow: {
-          workflowId: '',
-          workflowType: { name: '' },
-          taskQueue: { name: '' },
+          workflowId: workflowId,
+          workflowType: { name: workflowType },
+          taskQueue: { name: taskQueue },
         },
       },
     },
   };
 
-  Object.keys(values).forEach((key) => {
-    setBodyProperty(key, body, values[key]);
+  setBodySpec(body, spec, presets);
+
+  // Wait 2 seconds for create to get it on fetchAllSchedules
```

Comment:
Anything better better we can do here? We should start to look into a more generic dead zone handler soon, this kind of code has become pretty frequent.

Reply:
Yeah probably soon...but it's just a setTimeout. Not sure yet that we need to pull it out until we get a more complex pattern. I don't love it but rather keep these in their respective services until we can refactor it out into a more wholistic solution (which I'm not sure what it will be yet).